packages/swapper/src/manager/SwapperManager.ts

```diff
@@ -89,44 +91,29 @@ export class SwapperManager {
       buyAssetId: buyAsset.assetId,
     })
 
-    // Get quotes from all swappers that support the pair
-    const quotePromises: Promise<SwapperQuoteTuple>[] = supportedSwappers.map(
-      async (swapper) => [swapper, await swapper.getTradeQuote(args)] as const,
-    )
-    const settledQuoteRequests = await Promise.allSettled(quotePromises)
-    // For each swapper, get receive amount/(input amount + gas fee), where all values are in fiat
-    const fulfilledQuoteTuples = settledQuoteRequests
-      .filter(isFulfilled)
-      .map((quoteRequest) => quoteRequest.value)
-
-    // The best swapper is the one with the highest ratio
-    const bestQuoteTuple = await fulfilledQuoteTuples.reduce(
-      async (acc: Promise<SwapperRatioTuple> | undefined, currentQuoteTuple: SwapperQuoteTuple) => {
-        const resolvedAcc = await acc
-        const [currentSwapper, currentQuote] = currentQuoteTuple
-        const currentRatio = await getRatioFromQuote(currentQuote, currentSwapper, feeAsset)
-
-        // It's our first iteration, so we just return the current SwapperQuoteTuple
-        if (!resolvedAcc) return Promise.resolve([currentSwapper, currentRatio])
-
-        const [, bestRatio] = resolvedAcc
-        const isCurrentSwapperBestSwapper = (() => {
-          // Happy path - no div by 0's in getRatioFromQuote evaluations so we have both ratios
-          if (!isUndefined(currentRatio) && !isUndefined(bestRatio)) return currentRatio > bestRatio
-          // We don't know, neither has a ratio, so we can't compare (big edge case div by 0 scenario)
-          if (isUndefined(currentRatio) && isUndefined(bestRatio)) return false
-          // We don't have a best, but we do have a current, so current is the new best
-          return !isUndefined(currentRatio)
-        })()
-
-        const currentSwapperRatioTuple = [currentSwapper, currentRatio] as const
-        return Promise.resolve(isCurrentSwapperBestSwapper ? currentSwapperRatioTuple : resolvedAcc)
-      },
-      undefined,
-    )
-
-    const bestSwapper = bestQuoteTuple?.[0]
-    return bestSwapper
+    const settledSwapperDetailRequests: PromiseSettledResult<SwapperWithQuoteMetadata>[] =
+      await Promise.allSettled(
+        supportedSwappers.map(async (swapper) => {
+          const quote = await swapper.getTradeQuote(args)
```

Comment:
As a safety net, do we want to catch errors here and `console.error()` instead of throwing? I.e no quote but still a valid swapper

Reply:
As discussed out of band - our current assumption is that if we can't get a quote, the swapper is not valid for that pair/input args.